From: Long Li <longli@microsoft.com>
Date: Mon, 12 Sep 2016 16:54:01 -0700
Subject: pci-hyperv: properly handle device eject
Patch-mainline: submitted (lkml - https://lkml.org/lkml/2016/9/12/1129)
References: fate#320295, bnc#998664

A PCI_EJECT message can arrive at the same time we are calling
pci_scan_child_bus in the workqueue for the previous PCI_BUS_RELATIONS
message, in this case we could potentailly modify the bus from two
places. Properly lock the bus access.

Signed-off-by: Long Li <longli@microsoft.com>
Acked-by: <ohering@suse.de>
---
 drivers/pci/host/pci-hyperv.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pci/host/pci-hyperv.c b/drivers/pci/host/pci-hyperv.c
index f6a7417..06aae25 100644
--- a/drivers/pci/host/pci-hyperv.c
+++ b/drivers/pci/host/pci-hyperv.c
@@ -1591,7 +1591,7 @@ static void hv_eject_device_work(struct work_struct *work)
 	pdev = pci_get_domain_bus_and_slot(hpdev->hbus->sysdata.domain, 0,
 					   wslot);
 	if (pdev) {
-		pci_stop_and_remove_bus_device(pdev);
+		pci_stop_and_remove_bus_device_locked(pdev);
 		pci_dev_put(pdev);
 	}
 
